lib/jekyll/post.rb

```diff
@@ -118,6 +119,13 @@ def title
       data.fetch('title') { titleized_slug }
     end
 
+    # Public: the Post excerpt_separator, from the YAML Front-Matter or empty string
+    #
+    # Returns the post excerpt_separator
+    def excerpt_separator
+      data.fetch('excerpt_separator') { "" }
```

Comment:
I mean instead of just returning `""` straight away, it should check the defaults and the site first.